pyproject.toml

```diff
@@ -1,11 +1,143 @@
 [build-system]
-# Minimum requirements for the build system to execute.
-requires = ["setuptools>=61.0.0", "setuptools_scm>=7.0.1"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.0.0", "hatch-vcs"]
+build-backend = "hatchling.build"
 
-[tool.setuptools_scm]
-write_to = "src/pyhf/_version.py"
-local_scheme = "no-local-version"
+[project]
+name = "pyhf"
+dynamic = ["version"]
+description = "pure-Python HistFactory implementation with tensors and autodiff"
+readme = "README.rst"
```

Comment:
Do we want to use https://github.com/hynek/hatch-fancy-pypi-readme#readme to remove the badges here? We use it in Awkward, for reference.

Reply:
Which badges get removed? I haven't looked at how `hatch-fancy-pypi-readme` works yet but comparing https://github.com/scikit-hep/awkward/tree/b6e4c71029152fcd93c1b156d8d4dfb8227b1e24 and https://pypi.org/project/awkward/2.0.6/ it looks like the things stripped out is from "Installation for developers" downwards.

Reply:
You're totally right! I had a memory of removing the badges too, but I think Jim preferred them in there. 

We use the same mechanism though — one can add hidden markers to the `README.rst` / `README.md`, and treat them as boundaries of text chunks that are used to build the final README.

Reply:
Moved to Issue #2097